commands/bake.go

```diff
@@ -134,7 +135,18 @@ func runBake(dockerCli command.Cli, targets []string, in bakeOptions, cFlags com
 	}()
 
 	if url != "" {
-		files, inp, err = bake.ReadRemoteFiles(ctx, nodes, url, in.files, printer)
+		if containsLocalFiles(in.files) {
+			var remoteFiles []bake.File
+			remoteFiles, inp, err = bake.ReadRemoteFiles(ctx, nodes, url, getRemoteFiles(in.files), printer)
+			if err != nil {
+				return err
+			}
+			files, err = bake.ReadLocalFiles(getLocalFiles(in.files))
+			files = append(files, remoteFiles...)
+
+		} else {
+			files, inp, err = bake.ReadRemoteFiles(ctx, nodes, url, in.files, printer)
```

Comment:
Avoid duplicating this call. Instead split the files list into remote and local files with a single function and always call `ReadLocalFiles/ReadRemoteFiles` if array has items.